test/promise-subclass.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = (tap) => {
+  return class SubclassedPromise extends Promise {
+    then(onSuccess, onReject) {
+      tap.type(onSuccess, 'function');
+      tap.type(onReject, 'undefined');
+      return Promise.prototype.then.call(this, onSuccess, onReject);
```

Comment:
I'm not sure if it makes a difference in this case as I'm not that familiar with the new class syntax, but is there a difference between `Promise.prototype.then.call()` and `super.then()`? I haven't seen the two styles mixed like this before